src/java.base/share/classes/java/util/concurrent/ThreadLocalRandom.java

```diff
@@ -1066,22 +422,55 @@ private Object readResolve() {
     private static final long INHERITEDACCESSCONTROLCONTEXT
         = U.objectFieldOffset(Thread.class, "inheritedAccessControlContext");
 
-    /** Rarely-used holder for the second of a pair of Gaussians */
-    private static final ThreadLocal<Double> nextLocalGaussian =
-        new ThreadLocal<>();
-
     /** Generates per-thread initialization/probe field */
     private static final AtomicInteger probeGenerator = new AtomicInteger();
 
     /** The common ThreadLocalRandom */
     static final ThreadLocalRandom instance = new ThreadLocalRandom();
 
+    private static final class ThreadLocalRandomProxy extends Random {
+        @java.io.Serial
+        static final long serialVersionUID = 0L;
+
+       public int nextInt() {
+           return ThreadLocalRandom.current().nextInt();
+       }
+
+       public long nextLong() {
+           return ThreadLocalRandom.current().nextLong();
+       }
+    }
+
+    private static final AbstractSpliteratorGenerator proxy = new ThreadLocalRandomProxy();
+
+    /**
+     * Returns a {@code RandomGenerator} object that uses {@code ThreadLocalRandom}.
+     * This object is threadsafe (methods of this object may be called by threads
+     * other than the current one).
+     *
+     * @return a {@code RandomGenerator} object that uses {@code ThreadLocalRandom}
+     */
+    public static RandomGenerator proxy() {
+        return proxy;
+    }
+
+    // Methods required by class AbstractSpliteratorGenerator
+    public Spliterator.OfInt makeIntsSpliterator(long index, long fence, int origin, int bound) {
+        return new RandomIntsSpliterator(proxy, index, fence, origin, bound);
```

Comment:
should use proxy(=

Reply:
will investigate

Reply:
Needed to use ThreadLocalRandomProxy.proxy otherwise a cast would be required.

Reply:
yes, right !